docs/s3.md

```diff
@@ -0,0 +1,153 @@
+# S3
+
+Action uses Amazon S3 for three purposes:
+
+   1. Serving manually managed assets, such as logos
+   2. Serving builds bundled with webpack
+   3. Storing and serving user-generated assets
+
+Action uses a single S3 bucket. Within the bucket, the directory structure
+is as follows:
+
+```
+   S3 Bucket
+   =========
+   |
+   |--- static/                     1) Manually managed assets
+   |
+   |--- :instance/
+      |--- build/                   2) Builds bundled with webpack
+      |  |--- :vX.Y.Z/
+      |
+      |--- store/                   3) User-generated assets
```

Comment:
I went with a compromise here. The teamId _is_ deterministic but the filename is not. Also, listing on the team directory is disabled. So, admins (us) can see files associated with a given object easily, but users can't just walk the S3 store.

If shortId isn't long enough, we can always use a longer filename...

What do you think?